tls/s2n_crypto.h

```diff
@@ -72,6 +72,8 @@ struct s2n_crypto_parameters {
     struct s2n_ecc_evp_params server_ecc_evp_params;
     struct s2n_ecc_evp_params client_ecc_evp_params[S2N_ECC_EVP_SUPPORTED_CURVES_COUNT];
     const struct s2n_ecc_named_curve * mutually_supported_groups[S2N_ECC_EVP_SUPPORTED_CURVES_COUNT];
+    struct s2n_ecc_named_curve preferred_key_shares[S2N_ECC_EVP_SUPPORTED_CURVES_COUNT];
```

Comment:
have you thought about making this an s2n_array?

Reply:
I did. The reason I chose against it was memory management overhead and order or curves. If we use the s2n_array, then we have to search the whole array every time to see if we find a match. If we know that the order is the same as the other lists, we can use the same index.